data_clustering/clustering_functions.py

```diff
@@ -0,0 +1,24 @@
+from typing import Callable
+# Function that receives a list of chess moves and returns a list of centroids
+ClusteringStategyFunction = Callable[[list[tuple[int, int]]], list[float]]
+
+# Function that receives 2 chess moves and returns their difference
+ChessDistanceFunction = Callable[[tuple[int,int], tuple[int,int]], float]
+
+def k_means_clustering(move_list: list[tuple[int,int]], number_of_clusters: int) -> list[float]:
```

Comment:
de obicei daca ai functii separate nu este prea ok, mai bine definesti o clasa abstracta cu functiile necesare(ideal ar fi interfete dar in python nu prea exista) si extinzi acelea cu implementari, in cazul nostru:
1.Ai o calsa de clustering cu functia "clustering"(sau alta nume) cu acesta intrari si iesiri
2.Ai a clasa de "Distance" cu functia "getDistance"(sau ceva de gen) cu intrari si iesiri cum este la la mandoua distante
ca si implementare extinzi aceste clase

Reply:
daca nu ai timp, poti sa ignori